### PR TITLE
refactor: add apic api update example

### PR DESCRIPTION
**Pull request by frankqianms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/api/_update.py [hunk 1]
```diff
@@ -22,6 +22,9 @@ class Update(AAZCommand):
 
     :example: Update custom properties
         az apic api update -g contoso-resources -n contoso --api-id echo-api --custom-properties '{\"public-facing\":true}'
+
+    :example: Update single custom metadata
+        az az apic api update -g contoso-resources -n contoso --api-id echo-api --set customProperties.internal=false
```
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/aaz/latest/apic/api/_update.py:27]: Please remove the duplicate `az` in the example. `az` is added by default so you don't need to include it when authoring samples in aaz-dev

Comment (frankqianms) [reply src/apic-extension/azext_apic_extension/aaz/latest/apic/api/_update.py]: removed

File: src/apic-extension/azext_apic_extension/tests/latest/test_api_commands.py [hunk 2]
```diff
@@ -184,6 +184,18 @@ def test_examples_update_api(self):
...
 
+    @ResourceGroupPreparer(name_prefix="clirg", location=TEST_REGION, random_name_length=32)
+    @ApicServicePreparer()
+    @ApicApiPreparer()
+    @ApicMetadataPreparer()
+    def test_examples_update_single_custom_metadata(self, metadata_name):
+        self.kwargs.update({
+            'customProperty': '{}'.format(metadata_name),
+        })
+        self.cmd('az apic api update -g {rg} -n {s} --api-id {api} --set customProperties.{customProperty}=false', checks=[
```
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/tests/latest/test_api_commands.py:195]: Minor: The `ApicMetadataPreparer` sets the metadata name to kwargs `m` (see https://github.com/blackchoey/azure-cli-extensions/blob/main/src/apic-extension/azext_apic_extension/tests/latest/utils.py#L53).

You can change the code to `self.cmd('az apic api update -g {rg} -n {s} --api-id {api} --set customProperties.{m}=false'` and remove the `self.kwargs.update(...)` code to simplify the test case.

Comment (frankqianms) [reply src/apic-extension/azext_apic_extension/tests/latest/test_api_commands.py]: very good suggestion. optimized

**Review by blackchoey — APPROVED** [review]

**Pull request merged**